### PR TITLE
[lldb][Docs] Convert platform packets doc to Markdown

### PR DESCRIPTION
**Pull request by DavidSpickett**
As before, script did most of the work, hand edits after that.

There's a lot more we can do dedupe this and the packets doc, this will come in a follow up PR.

### CODE REVIEW

**Review by JDevlieghere — APPROVED** [review]
LGTM. 

PS: I was looking at the website and I wonder if we should remove all the "Brief" subtitles and put the brief description right under the packet name. I don't think the subtitle adds much. If you agree, this can be done in a separate PR before or after this one.

File: lldb/docs/resources/lldbplatformpackets.md [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+### Example
+
+```
+receive: qPlatform_mkdir:000001fd,2f746d702f6131
+send:    F0
+```
+
+request packet has the fields:
+   1. mode bits in base 16
+   2. file path in ascii-hex encoding
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:107]:
```suggestion
   2. file path in ASCII hex encoding
```
(for consistency)

File: lldb/docs/resources/lldbplatformpackets.md [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+
+```
+receive: qPlatform_shell:6c73202f746d702f,0000000a
+send:    F,0,0,<OUTPUT>
+```
+
+request packet has the fields:
+   1. shell command ascii-hex encoded
+   2. timeout
+   3. working directory ascii-hex encoded (optional)
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:128]:
```suggestion
   3. working directory in ASCII hex encoding (optional)
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+on the target system.
+
+### Example
+
+```
+receive: qLaunchGDBServer;host:<HOSTNAME_LLDB_IS_ON>;
+send:    pid:1337;port:43001;
+```
+
+Request packet hostname field is not ascii-hex encoded. Hostnames
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:148]:
```suggestion
Request packet hostname field is not ASCII hex encoded. Hostnames
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+receive: qKillSpawnedProcess:1337
+send:    OK
+```
+The request packet has the process ID in base 10.
+
+## qProcessInfoPID:
+
+### Brief
+
+Gather information about a process running on the target
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:175]:
```suggestion
Gather information about a process running on the target.
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+
+### Example
+
+```
+receive: qPlatform_shell:6c73202f746d702f,0000000a
+send:    F,0,0,<OUTPUT>
+```
+
+request packet has the fields:
+   1. shell command ascii-hex encoded
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:126]:
```suggestion
   1. shell command in ASCII hex encoding
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+| `triple`     | ascii-hex | An ASCII hex target triple string (`x86_64`, `x86_64-apple-macosx`, `armv7-apple-ios`)
+
+If no criteria is given, `qfProcessInfo` will request a list of every process.
+
+The lldb testsuite currently only uses `name_match:equals` and the
+no-criteria mode to list every process.
+
+The response should include any information about the process that
+can be retrieved in semicolon-separated `name:value` fields.
+In this example, `pid` is base 10, `name` is ascii-hex encoded.
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:228]:
```suggestion
In this example, `pid` is base 10, `name` is ASCII hex encoded.
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+receive: qPathComplete:1,746573
+send:    M746573742f,74657374732f
+```
+
+If the first argument is zero, the result should contain all
+files (including directories) starting with the given path. If the
+argument is one, the result should contain only directories.
+
+The result should be a comma-separated list of hex-encoded paths.
+Paths denoting a directory should end with a directory separator (`/` or `\`.
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:279]:
```suggestion
Paths denoting a directory should end with a directory separator (`/` or `\`).
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+
+## vFile:size
+
+### Brief
+
+Get the size of a file on the target system, filename in ASCII hex.
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:285]:
```suggestion
Get the size of a file on the target system, filename in ASCII hex encoding.
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+```
+
+response is `F` followed by the file size in base 16.
+`F-1,errno` with the errno if an error occurs, base 16.
+
+## vFile:mode
+
+### Brief
+
+Get the mode bits of a file on the target system, filename in ASCII hex.
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:301]:
```suggestion
Get the mode bits of a file on the target system, filename in ASCII hex encoding.
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+### Example
+
+```
+receive: vFile:unlink:2f746d702f61
+send:    F0
+```
+
+Argument is a file path in ascii-hex encoding.
+Response is `F` plus the return value of `unlink()`, base 16 encoding.
+Return value may optionally be followed by a comma and the base16
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:329]:
```suggestion
Return value may optionally be followed by a comma and the the value of errno, in base 16 encoding, if unlink failed.
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 11]
```diff
@@ -0,0 +1,482 @@
...
+value of errno if unlink failed.
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:330]:
```suggestion

```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 12]
```diff
@@ -0,0 +1,482 @@
...
+receive: vFile:pread:7,1024,0
+send:    F4;a'b\00
+```
+
+Request packet has the fields:
+   1. File descriptor, base 16
+   2. Number of bytes to be read, base 16
+   3. Offset into file to start from, base 16
+
+Response is `F`, followed by the number of bytes read (base 16), a
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:443]:
```suggestion
Response is `F`, followed by the number of bytes read (base 16 encoded), a
```

File: lldb/docs/resources/lldbplatformpackets.md [hunk 13]
```diff
@@ -0,0 +1,482 @@
...
+receive: vFile:pwrite:8,0,\cf\fa\ed\fe\0c\00\00
+send:    F1024
+```
+
+Request packet has the fields:
+   1. File descriptor, base 16
+   2. Offset into file to start from, base 16
+   3. binary-escaped-data to be written
+
+Response is `F`, followed by the number of bytes written (base 16).
```
Comment (hawkinsw) [inline lldb/docs/resources/lldbplatformpackets.md:464]:
```suggestion
Response is `F`, followed by the number of bytes written (base 16 encoded).
```

**Review by hawkinsw — COMMENTED** [review]
I hope that these little fixes are helpful! I think that the documentation is really thorough and helpful!

**Pull request merged**